File: packages/android/core-domain/src/test/kotlin/app/notes/domain/TagTest.kt
package app.notes.domain

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test
import java.time.Instant

class TagTest {
    private val owner = UserId("550e8400-e29b-41d4-a716-446655440000")
    private val id = TagId("550e8400-e29b-41d4-a716-446655440003")
    private val now: Instant = Instant.parse("2026-05-20T12:00:00Z")

    @Test
    fun `accepts a valid tag`() {
        val tag = Tag(id, owner, "work", "#112233", now, now)
        assertEquals("work", tag.name)
    }

    @Test
    fun `rejects a blank name`() {
        assertThrows(IllegalArgumentException::class.java) {
            Tag(id, owner, "", "#112233", now, now)
        }
    }

    @Test
    fun `rejects a malformed color`() {
        assertThrows(IllegalArgumentException::class.java) {
            Tag(id, owner, "work", "#xyz", now, now)
        }
    }
}

File: packages/domain/src/__tests__/tag.test.ts
import { describe, expect, it } from "vitest";
import { makeTag, type Tag } from "../tag.js";
import { TagId, UserId } from "../ids.js";

const OWNER = UserId("550e8400-e29b-41d4-a716-446655440000");
const TAG = TagId("550e8400-e29b-41d4-a716-446655440003");

function base(overrides: Partial<Tag> = {}): Tag {
  return {
    id: TAG,
    ownerId: OWNER,
    name: "work",
    color: "#112233",
    createdAt: "2026-05-20T12:00:00.000Z",
    updatedAt: "2026-05-20T12:00:00.000Z",
    ...overrides,
  };
}

describe("makeTag", () => {
  it("accepts a valid tag", () => {
    expect(makeTag(base()).name).toBe("work");
  });

  it("rejects a blank name", () => {
    expect(() => makeTag(base({ name: "" }))).toThrow(/name/);
  });

  it("rejects a malformed color", () => {
    expect(() => makeTag(base({ color: "#xyz" }))).toThrow(/color/);
  });
});
